File: src/main/java/io/github/kingg22/api/vacunas/panama/web/dto/DoctorDto.java
package io.github.kingg22.api.vacunas.panama.web.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.kingg22.api.vacunas.panama.persistence.entity.Doctor;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PastOrPresent;
import jakarta.validation.constraints.Size;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

/**
 * DTO for {@link Doctor}
 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@FieldDefaults(level = AccessLevel.PRIVATE)
public class DoctorDto extends PersonaDto implements Serializable {
    @NotNull
    @Size(max = 20)
    String idoneidad;

    @Size(max = 100)
    String categoria;

    @PastOrPresent
    @JsonProperty(value = "created_at")
    LocalDateTime createdAt;

    @PastOrPresent
    @JsonProperty(value = "updated_at")
    LocalDateTime updatedAt;
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/web/dto/DoctorDto.kt
package io.github.kingg22.api.vacunas.panama.web.dto

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.PastOrPresent
import jakarta.validation.constraints.Size
import java.io.Serializable
import java.time.LocalDateTime

/** DTO for [io.github.kingg22.api.vacunas.panama.persistence.entity.Doctor]  */
class DoctorDto @JvmOverloads constructor(
    @Size(max = 20)
    val idoneidad: String? = null,

    @Size(max = 100)
    val categoria: String? = null,

    @JsonProperty(value = "created_at")
    @PastOrPresent
    val createdAt: LocalDateTime? = null,

    @JsonProperty(value = "updated_at")
    @PastOrPresent
    val updatedAt: LocalDateTime? = null,
) : PersonaDto(),
    Serializable
